crates/services/block_aggregator_api/proto/api.proto

```diff
@@ -14,9 +14,519 @@ message BlockRangeRequest {
+//    /// Generated application fields.
+//    pub generated: Generated,
+//}
+// pub struct GeneratedApplicationFieldsV1 {
+//    /// Number of transactions in this block.
+//    pub transactions_count: u16,
+//    /// Number of message receipts in this block.
+//    pub message_receipt_count: u32,
+//    /// Merkle root of transactions.
+//    pub transactions_root: Bytes32,
+//    /// Merkle root of message receipts in this block.
+//    pub message_outbox_root: Bytes32,
+//    /// Root hash of all imported events from L1
+//    pub event_inbox_root: Bytes32,
+//}
+// pub struct ConsensusHeader<Generated> {
+//    /// Merkle root of all previous block header hashes.
+//    pub prev_root: Bytes32,
+//    /// Fuel block height.
+//    pub height: BlockHeight,
+//    /// The block producer time.
+//    pub time: Tai64,
+//    /// generated consensus fields.
+//    pub generated: Generated,
+//}
+// pub struct GeneratedConsensusFields {
+//    /// Hash of the application header.
+//    pub application_hash: Bytes32,
+//}
+// pub struct BlockHeaderMetadata {
+//    /// Hash of the header.
+//    id: BlockId,
+//}
+message V1Header {
+  uint32 da_height = 1;
+  uint32 consensus_parameters_version = 2;
+  uint32 state_transition_bytecode_version = 3;
+  uint32 transactions_count = 4;
+  uint32 message_receipt_count = 5;
+  bytes transactions_root = 6;
+  bytes message_outbox_root = 7;
+  bytes event_inbox_root = 8;
+  bytes prev_root = 9;
+  uint32 height = 10;
+  bytes time = 11;
+  bytes application_hash = 12;
+  optional bytes block_id = 13;
+}
+
+// pub struct GeneratedApplicationFieldsV2 {
+//    /// Number of transactions in this block.
+//    pub transactions_count: u16,
+//    /// Number of message receipts in this block.
+//    pub message_receipt_count: u32,
+//    /// Merkle root of transactions.
+//    pub transactions_root: Bytes32,
+//    /// Merkle root of message receipts in this block.
+//    pub message_outbox_root: Bytes32,
+//    /// Root hash of all imported events from L1
+//    pub event_inbox_root: Bytes32,
+//    /// TxID commitment
+//    pub tx_id_commitment: Bytes32,
+//}
+message V2Header {
+  uint32 da_height = 1;
+  uint32 consensus_parameters_version = 2;
+  uint32 state_transition_bytecode_version = 3;
+  uint32 transactions_count = 4;
+  uint32 message_receipt_count = 5;
+  bytes transactions_root = 6;
+  bytes message_outbox_root = 7;
+  bytes event_inbox_root = 8;
+  bytes tx_id_commitment = 9;
+  bytes prev_root = 10;
+  uint32 height = 11;
+  bytes time = 12;
+  bytes application_hash = 13;
+  optional bytes block_id = 14;
+}
+
+// pub enum Transaction {
+//    Script(Script),
+//    Create(Create),
+//    Mint(Mint),
+//    Upgrade(Upgrade),
+//    Upload(Upload),
+//    Blob(Blob),
+//}
+message Transaction {
+  oneof variant {
+    ScriptTx script = 1;
+//    CreateTx create = 2;
+//    MintTx mint = 3;
+//    UpgradeTx upgrade = 4;
+//    UploadTx upload = 5;
+//    BlobTx blob = 6;
```

Comment:
Instead of these being `...Tx` can we spell out `Transaction`? It keeps it cleaner & had no impact to the traffic on the wire.